docs/memory-safety-continuum.md

```diff
@@ -0,0 +1,99 @@
+
+This DRAFT document lays out the idea of the "Memory Safety Continuum". Eventually, this idea may be used as a foundation for other content which will help developers and organizations identify where they are on the continuum and how to get to where they want to be.
+
+## Audience
+
+The audience for this draft is developers - however, the audience may expand with future drafts/versions of this document.
+
+## What is Memory Safety?
+
+Rather than using terms like "Memory Safe Language" and "Memory Unsafe Language", this SIG prefers the terms "memory safe by default" and "non-memory safe by default". Please see our [useful definitions](definitions.md) file for more information about memory safety and undefined behavior.
+
+## The Continuum
+
+This is a very rough idea of what the continuum might look like - from "least safety" to "most safety"
+
+* Using a non-memory safe by default language (such as C or C++) without developer best practices or automated tooling to check for memory safety
+* Using a non-memory safe by default language with developer best practices, but no automated tooling to check for memory safety
+* Using a non-memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code
+* Using a non-memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code AND automated tooling to check for memory safety in third party code (dependencies)
+* Using a memory safe by default language (such as Rust, Go, Python, Java, JavaScript, C#) without developer best practices and automated tooling
+* Using a memory safe by default language with developer best practices, but no automated tooling to check for memory safety
+* Using a memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code
+* Using a memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code AND automated tooling to check for memory safety in third party code (dependencies)
+
+### Using a non-memory safe by default language without developer best practies or automated tooling
+
+Using raw C or C++ (or old versions of C and C++ language and compiler)
+
+### Using a non-memory safe by default language with developer best practices, but no automated tooling
+
+Examples:
+
+* [Using attributes such as `cleanup` and classes when writing C](https://lwn.net/Articles/934679/) (and depending on developers to manually check this)
+* Following the [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines) when writing C++
+* Using the [C++ Compiler Hardening Guide](https://github.com/ossf/wg-best-practices-os-developers/tree/main/docs/Compiler-Hardening-Guides) when compiling C++ code
+
```

Comment:
How about? :

Isolating code that processes un-trusted data from code that performs direct memory management operations or uses raw pointers (See: https://langsec.org)